sql/core/src/main/scala/org/apache/spark/sql/catalog/Catalog.scala

```diff
@@ -226,4 +226,11 @@ abstract class Catalog {
    */
   def refreshTable(tableName: String): Unit
 
+  /**
+   * Invalidate and refresh all the cached data (and the associated metadata) for any dataframe that
+   * contains the given data source path.
+   *
+   * @since 2.0.0
+   */
+  def refreshResource(path: String): Unit
```

Comment:
Should we call it `invalidateCache()` to reflect the things we actually done?

Also, it's a bit confusing to have this API on `catalog`, can we put it on SparkSession?